src/Shapes/inertiaTensorAndVolume.jl

```diff
@@ -39,14 +39,14 @@ topArea(shape::Cone) = pi*(shape.topDiameter/2)^2
 Return the volume of the solid shape `shape::Modia3D.AbstractGeometry` in [m^3].
 """
 volume(shape::Modia3D.AbstractGeometry) = bottomArea(shape)*lengthGeo(shape)
-volume(shape::Sphere)    = 4/3*pi*(shape.diameter/2)^3
-volume(shape::Ellipsoid) = 4/3*pi*shape.lengthX/2*shape.lengthY/2*shape.lengthZ/2
-volume(shape::Cone)      = pi/12*shape.length*(shape.diameter^2 + shape.diameter*shape.topDiameter + shape.topDiameter^2)
-
-function volume(shape::Capsule)
-    r = shape.diameter/2
-    h = shape.length
-    return pi*r^2*h + 4/3*pi*r^3
+volume(shape::Sphere{F}) where {F} = F(4/3*pi*(shape.diameter/2)^3)
+volume(shape::Ellipsoid{F}) where{F} = F(4/3*pi*shape.lengthX/2*shape.lengthY/2*shape.lengthZ/2)
+volume(shape::Cone{F}) where {F} = F(pi/12*shape.length*(shape.diameter^2 + shape.diameter*shape.topDiameter + shape.topDiameter^2))
+
+function volume(shape::Capsule{F}) where {F}
+    r = F(shape.diameter/2)
```

Comment:
Is F(*) required here?

Reply:
Yes, it is. Default types are `Float64` and `Int64`. 
 If e.g. `Float64(3.5)*Float32(1.5)` the result is `Float64(5.25)`. 

Ok, in this case `shape.diameter` is of `Type{F}` and `2` is of `Type{Int64}`. Therefore, it would work because `Int64` is converted to the needed `Float`.
But if it would be `2.0` it's default type is `Float64` and it won't work.